src/bin/pg_autoctl/monitor.c

```diff
@@ -3843,7 +3845,23 @@ monitor_ensure_extension_version(Monitor *monitor,
 				 PG_AUTOCTL_MONITOR_EXTENSION_NAME,
 				 version->installedVersion);
 
-		return true;
+		/*
+		 * Now that we have done the ALTER EXTENSION UPDATE, our background
+		 * workers on the monitor have been started with the new shared library
+		 * object and the old SQL definitions. Let's restart Postgres so that
+		 * the background workers have a chance of a fresh start with an SQL
+		 * schema that matches the expectations of the shared library code.
+		 */
+		log_info("Restarting Postgres on the monitor");
+
+		if (!ensure_postgres_service_is_stopped(postgres))
```

Comment:
This seems to be causing this error in the monitor logs:
```
15:06:22 4598 INFO  Restarting Postgres on the monitor
15:06:22 4597 INFO  Stopping pg_autoctl postgres service
15:06:22 4597 INFO  /home/jelte/.pgenv/pgsql/bin/pg_ctl --pgdata /home/jelte/work/pg_auto_failover/monitor --wait stop --mode fast
15:06:22 4645 INFO   /home/jelte/.pgenv/pgsql/bin/postgres -D /home/jelte/work/pg_auto_failover/monitor -p 5000 -h *
15:06:22 4597 WARN  PostgreSQL was not running, restarted with pid 4645
15:06:22 4598 INFO  Contacting the monitor to LISTEN to its events.
15:06:22 4598 ERROR Failed to LISTEN "state": FATAL:  terminating connection due to administrator command
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

15:06:22 4598 WARN  Re-establishing connection. We might miss notifications.
```

It doesn't cause any real issues, so it's not much of a problem. If we can make sure it doesn't happen that would be better though. One idea would be te stop our LISTEN temporarily when we stop postgres.

Reply:
Good point, fixed now.